adding-a-database-to-your-app-using-okteto-stack.md

```diff
@@ -0,0 +1,369 @@
+    if not get_single_recipe(id):
+        return {
+            "error": "No such recipe exist"
+        }
+
+    update_recipe_data(id, recipe_data.dict())
+
+    return {
+        "message": "Recipe updated successfully."
+    }
+
+@app.delete("/recipe/{id}", tags=["Recipe"])
+def delete_recipe(id: str) -> dict:
+    if not get_single_recipe(id):
+        return {
+            "error": "Invalid ID passed"
+        }
+
+
+    remove_recipe(id)
+    return {
+        "message": "Recipe deleted successfully."
+    }
+```
+
+Update the `api/model.py` by removing the `id` field in the `RecipeSchema` model class.
+
+## Testing The Database
+
+With the database connection in place, start a `mongod` server to allow interactions with the database:
+
+```
+mongod --port 27017
+```
+
+Next, test the POST route:
+
+```bash
+(venv)$ curl -X POST http://localhost:8080/recipe -d \
+'{"name": "Donut", "ingredients": ["Flour", "Milk", "Butter"]}' \
+-H 'Content-Type: application/json'
+```
+
+Response:
+
+```json
+{
+  "id": "601fdcd82fbbf462d33a6e34"
+}
+```
+
+Test the GET routes:
+
+1. Return all recipes
+
+```bash
+(venv)$  curl -X GET http://localhost:8080/recipe/2 -H 'Content-Type: application/json'
+```
+
+Response:
+
+```json
+{
+  "data": [
+    {
+      "id": "601fdcd82fbbf462d33a6e34",
+      "name": "Donut",
+      "ingredients": [
+        "Flour",
+        "Milk",
+        "Butter"
+      ]
+    }
+  ]
+}
+```
+
+2. Return a single recipe
+
+```bash
+(venv)$ curl -X GET http://localhost:8080/recipe/601fdcd82fbbf462d33a6e34 -H 'Content-Type: application/json'
+```
+
+Response:
+
+```json
+{
+  "data": {
+    "id": "601fdcd82fbbf462d33a6e34",
+    "name": "Donut",
+    "ingredients": [
+      "Flour",
+      "Milk",
+      "Butter"
+    ]
+  }
+}
+```
+
+Test the UPDATE route:
+
+```bash
+(venv)$ curl -X PUT "http://0.0.0.0:8080/recipe?id=601fdcd82fbbf462d33a6e34" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{\"name\":\"Buns\",\"ingredients\":[\"Flour\",\"Milk\",\"Sugar\",\"Vegetable Oil\"]}"
+```
+
+Response:
+
+```json
+{
+  "message": "Recipe updated successfully."
+}
+```
+
+Lastly, test the DELETE route:
+
+```bash
+(venv)$ curl -X DELETE "http://0.0.0.0:8080/recipe/601fdcd82fbbf462d33a6e34" -H  "accept: application/json"
+```
+
+Response:
+
+```json
+{
+  "message": "Recipe deleted successfully."
+}
+```
+
+## Redeploying to Okteto
+
+Okteto eases the stress of deployment and subsequent redeployment by allowing us update and upgrade existing applications from the stack file. Include another service for mongodb in your stack file:
+
+```yaml
+  mongodb:
+    image: bitnami/mongodb:latest
+    ports:
+      - 27017
+    resources:
+      cpu: 100m
+      memory: 128Mi
+      storage: 1Gi
+    volumes:
+      - /bitnami/mongodb
+```
+
+In the code above, you added another service, `mongodb`, to house a MongoDB container from bitnami; the default port **27017** is exposed under the ports heading.
```

Comment:
Could you also mention that this time we are not using 'public' because we only want the API service to be able to access our mongodb instance?

Reply:
al explain that we added `volumes` to make our data persistent.

Reply:
Fixed